### PR TITLE
changelogs: Reflow line lengths + yaml cleanups

### LINKED ISSUE #26462: fix_format really noisy
it'd be lovely to silence the hundreds of lines of yamllint warnings for example as it can be hard to find early failures

CodeChecker WARNING [yamllint] changelogs/1.20.0.yaml (line-length 300:201): line too long (213 > 200 characters)

totally an nice-to-have and I'll let stalebot close it out if no one knows how to fix offhand (cc @phlax )

### PR DESCRIPTION
**Pull request by phlax**
- fix changelogs
- assorted cleanups
- set max yaml width to 140 (allowing for non-breaking words)
- make excess width an error

Fix #26462 

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
cc @alyssawilk i wasnt sure whether it was worth taking the time to write a script to reflow these or just do it by hand

after getting most of the way through one - i think a script will be necessary 8(

**Comment by phlax** [comment]
first attempt at script worked on recent versions - ill debug further ...

**Comment by phlax** [comment]
half wor
ked

File: changelogs/1.12.7.yaml [hunk 1]
```diff
@@ -3,19 +3,12 @@ date: September 29, 2020
 changes:
 - area: http
   change: |
-    fixed CVE-2020-25017. Previously header matching did not match on all headers for non-inline headers. This patch
-    changes the default behavior to always logically match on all headers. Multiple individual
-    headers will be logically concatenated with ``,`` similar to what is done with inline headers. This
-    makes the behavior effectively consistent. This behavior can be temporary reverted by setting
-    the runtime value ``envoy.reloadable_features.http_match_on_all_headers`` to ``false``.
-
-    Targeted fixes have been additionally performed on the following extensions which make them
```
Comment (phlax) [inline changelogs/1.12.7.yaml:12]: fix

File: changelogs/1.13.5.yaml [hunk 2]
```diff
@@ -3,26 +3,18 @@ date: September 29, 2020
 changes:
 - area: http
   change: |
-    fixed CVE-2020-25017. Previously header matching did not match on all headers for non-inline headers. This patch
```
Comment (phlax) [inline changelogs/1.13.5.yaml:6]: fix

File: changelogs/1.14.5.yaml [hunk 3]
```diff
@@ -3,26 +3,18 @@ date: September 29, 2020
 changes:
 - area: http
   change: |
-    fixed CVE-2020-25017. Previously header matching did not match on all headers for non-inline headers.
```
Comment (phlax) [inline changelogs/1.14.5.yaml:6]: fix

File: changelogs/1.15.1.yaml [hunk 4]
```diff
@@ -3,27 +3,18 @@ date: September 29, 2020
 changes:
 - area: http
   change: |
-    fixed CVE-2020-25017. Previously header matching did not match on all headers for non-inline
```
Comment (phlax) [inline changelogs/1.15.1.yaml:6]: fix

File: changelogs/1.20.0.yaml [hunk 5]
```diff
@@ -3,76 +3,64 @@ date: October 5, 2021
 behavior_changes:
 - area: config
   change: |
-    due to the switch to using work-in-progress annotations and warnings to indicate APIs
-    subject to change, the following API packages have been force migrated from ``v3alpha`` to ``v3``:
```
Comment (phlax) [inline changelogs/1.20.0.yaml:7]: fix

**Review by alyssawilk — APPROVED** [review]
you're the best!

Would have mildly preferred to keep the longer limit and have less churn but docs are yours and if you prefer 140 140 it is :-)

**Comment by phlax** [comment]
> Would have mildly preferred to keep the longer limit 

lets see how it goes - i think wherever we set the limit its going to be hit sometimes and this seemed reasonable in the context of existing docs/configs etc

we could do with an autofix - i had to edit a few of the changelogs by hand to handle lists etc correctly - not sure if i can auto that but will look

**Comment by phlax** [comment]
this is failing coverage - i think its a flake - im gonna push this through to avoid any more changelog conflict

**Pull request merged**